jdaviz/configs/imviz/plugins/viewers.py

```diff
@@ -238,6 +241,11 @@ def _get_real_xy(self, image, x, y, reverse=False):
                 # Convert X,Y from reference data to the one we are actually seeing.
                 # world_to_pixel return scalar ndarray that we need to convert to float.
                 if self.get_link_type(image.label) == 'wcs':
+                    bb = self.state.reference_data.coords._orig_bounding_box
+                    if bb is not None:
+                        ints = bb.intervals
```

Comment:
Pretty sure this is GWCS specific, so the `if` should check if `.coords` is actually a GWCS or not.

Reply:
what part is GWCS specific, `intervals` or just the existence of the bounding box?  I could change `if bb is not None` to `if hasattr(bb, 'intervals')` - would that work?

Reply:
You already brute-force the bounding_box assignment in parser, so that is going to exist whether it is GWCS or FITS WCS. But intervals is very specific to GWCS. It is actually in `astropy.modeling` that only GWCS uses, not FITS WCS.

Reply:
We cannot guarantee that anything that tries to ducktype GWCS would use intervals properly, so it is safer to check for `isinstance(something.coords, GWCS)`.

Reply:
is there an alternative/more general way to check if a pixel is within the bounding box?

Reply:
(the parser now only clears `bounding_box` and assigns `_orig_bounding_box` for GWCS objects, with a note in the code that extending that to other WCS objects would also need a matching implementation here)